src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -419,4 +475,19 @@ private boolean isExactSearchThresholdSettingSet(int filterThresholdValue) {
     private boolean canDoExactSearchAfterANNSearch(final int filterIdsCount, final int annResultCount) {
         return filterWeight != null && filterIdsCount >= knnQuery.getK() && knnQuery.getK() > annResultCount;
     }
+
+    enum FilterIdsSelectorType {
```

Comment:
add java docs and move this out.

Reply:
Fixed at https://github.com/opensearch-project/k-NN/pull/1402/commits/a82e59fa2050bdd9dcfa80c3b43578ae5d265256 i create a new class `FilterIdsSelector`